File: 같은숫자는싫어_LV1.cpp
#include <vector>
#include <iostream>
#include <stack>

using namespace std;

vector<int> solution(vector<int> arr) {
    vector<int> answer;
    stack<int> st;

    for (int a: arr) {
        if (st.empty() || st.top() != a) {
            st.push(a);
            answer.push_back(a);
        }
    }

    return answer;
}

File: 같은숫자는싫어_LV1.java
import java.util.*;

public class Solution {
    public int[] solution(int []arr) {
       Stack<Integer> st = new Stack<>();

        for(int i = 0; i < arr.length; i++){
            if(st.size() == 0){
                st.push(arr[i]);
            }
            if(st.peek() == arr[i]){
                continue;
            }
            st.push(arr[i]);
        }

         int[] answer = new int[st.size()];

        for(int i = st.size()-1; i >=0; i--){
            answer[i] = st.peek();
            st.pop();
        }

        return answer;
    }
}